core/commands/storage/hosts.go

```diff
@@ -63,6 +64,17 @@ func GetHostsFromDatastore(ctx context.Context, node *core.IpfsNode, mode string
 	return hosts, nil
 }
 
+func GetHostStorageConfig(node *core.IpfsNode) (nodepb.Node_Settings, error) {
+	var ns nodepb.Node_Settings
+	rds := node.Repo.Datastore()
+	b, err := rds.Get(GetHostStorageKey(node.Identity.Pretty()))
+	if err != nil && err != ds.ErrNotFound {
+		return ns, fmt.Errorf("cannot get selfKey: %s", err.Error())
+	}
+	err = ns.Unmarshal(b)
+	return ns, err
```

Comment:
better to return `(*nodepb.Note_Settings, error)` so you can do `if err != nil { return nil, err }`

Reply:
good point, thx